pkg/sql/ir_generator.go

```diff
@@ -125,30 +125,30 @@ func generateTrainStmt(slct *parser.SQLFlowSelectStmt) (*ir.TrainStmt, error) {
 	return trainStmt, nil
 }
 
-func loadModelMeta(pr *parser.SQLFlowSelectStmt, db *database.DB, cwd, modelDir, modelName string) (*parser.SQLFlowSelectStmt, error) {
+func loadModelMeta(pr *parser.SQLFlowSelectStmt, db *database.DB, cwd, modelDir, modelName string) (*parser.SQLFlowSelectStmt, *parser.SQLFlowSelectStmt, error) {
 	modelURI := modelName
 	if modelDir != "" {
 		modelURI = fmt.Sprintf("file://%s/%s", modelDir, modelName)
 	}
 
 	m, e := model.Load(modelURI, cwd, db)
 	if e != nil {
-		return nil, fmt.Errorf("load %v", e)
+		return nil, nil, fmt.Errorf("load %v", e)
 	}
 	// Parse the training SELECT statement used to train
 	// the model for the prediction.
 	tr, e := parser.ParseOneStatement(db.DriverName, m.TrainSelect)
 	if e != nil {
-		return nil, fmt.Errorf("parse: TrainSelect %v raise %v", m.TrainSelect, e)
+		return nil, nil, fmt.Errorf("parse: TrainSelect %v raise %v", m.TrainSelect, e)
 	}
 
 	if e := verifier.VerifyColumnNameAndType(tr.SQLFlowSelectStmt, pr, db); e != nil {
-		return nil, fmt.Errorf("VerifyColumnNameAndType: %v", e)
+		return nil, nil, fmt.Errorf("VerifyColumnNameAndType: %v", e)
 	}
 
 	pr.TrainClause = tr.TrainClause
 
-	return pr, nil
+	return pr, tr.SQLFlowSelectStmt, nil
```

Comment:
Does this change fix https://github.com/sql-machine-learning/sqlflow/issues/1759 ?

Reply:
Not really, this is only a feature requested from @Echo9573 to support some time series model training, please refer to: #1715